app/jobs/one_time_jobs/backfill_contract_parties.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module OneTimeJobs
+  class BackfillContractParties < ApplicationJob
+    def perform
+      Contract.sent_with_docuseal.find_each do |contract|
+        submitters = contract.docuseal_document["submitters"]
+        next if submitters.nil?
+
+        # Collect all emails from submitters in this contract
+        emails = submitters.map { |s| s["email"] }.compact.uniq
+        users_by_email = User.where(email: emails).index_by(&:email)
+
+        submitters.each do |submitter|
+          role = case submitter["role"]
+                 when "Contract Signee"
+                   "signee"
+                 when "Cosigner"
+                   "cosigner"
+                 when "HCB"
+                   "hcb"
+                 else
+                   nil
+                 end
+          next unless role.present?
+
+          email = submitter["email"]
+          user = users_by_email[email]
+
+          party = nil
+          begin
+            party = if user.present?
```

Comment:
We don't wanna tie a user to a cosigner.

```suggestion
            party = if user.present? && role != "cosigner"
```